api/base/elasticsearch_dsl_views.py

```diff
@@ -52,6 +100,24 @@ def get_default_search(self) -> edsl.Search | None:
     #       it works fine with default pagination
 
     # override rest_framework.generics.GenericAPIView
+    @property
+    def pagination_class(self):
+        """
+        When downloading a file assume no pagination is necessary unless the user specifies
+        """
+        is_file_download = any(
+            self.request.accepted_renderer.format == renderer.format
+            for renderer in self.FILE_RENDERER_CLASSES
+        )
+
+        direct_download = self.request.query_params.get('direct_download')  # preserve pagination with direct download
+
+        page_size_param = getattr(super().pagination_class, 'page_size_query_param', 'page[size]')
+        # if it's a direct download of the JSON respect default page size
+        if is_file_download and not self.request.query_params.get(page_size_param) and not direct_download:
```

Comment:
bit confused by the logic around page size -- if there's a request with `format=tsv&direct_download=yes` it'll get 10 rows (and `format=tsv&page[size]=10000` will get 100 rows), which doesn't seem to match your intent

might be simpler if you:
- require explicit `page[size]=10000` for the full download (low default page size but high max page size)
- get rid of `FILE_RENDERER_CLASSES`; let the `direct_download` param (alone) control whether `Content-Disposition: attachment` gets set (and perhaps whether `page[size]=10000` is allowed, if you're worried about that)

seems reasonable to make the frontend specify `format=tsv&page[size]=10000&direct_download=pls` when we want all those things, rather than trying to infer it all by renderer choice

Reply:
Yes of course if you download a tsv it doesn't paginate it, it ignores that param, that's intentional. Maybe you can explain more.

Reply:
the issue is how the query params interact -- if you add `direct_download=true` to a request that has `format=tsv`, your logic as written will use default pagination and return only 10 rows

you might be able to tweak this line to handle that case, but i'm also suggesting it may be easier overall to make a coherent api if you think of the download as one big page and give the `direct_download` param consistent behavior for any renderer

Reply:
Why would the user do that?

Reply:
I'm confused, why would a user try to `direct_download=pls` a tsv? If go to Google and I add `direct_download=pls` that doesn't do anything.

Reply:
I also don't understand why you would paginate a tsv, but it's true that it doesn't make sense to necessarily change pagination based on renderer. 

I think having `direct_download=true` or whatever is your truthy parameter is an acceptable way of making this consistent and explicit. So if your `direct_download` is truthy, download everything rather than paginating it.

I don't like requiring a `page[size]` parameter to download everything because there's no guarantee what the page size has to be without making at least one additional request.

Reply:
We're running behind so I remove the direct JSON feature. Anymore issues?

Reply:
@brianjgeiger I'm merging this. He accepted the changes.